stake-pool/cli/src/main.rs

```diff
@@ -143,299 +230,343 @@ fn command_create_account(config: &Config, token: Pubkey) -> CommandResult {
+                &config.owner.pubkey(),
+                &[],
+                tokens_to_burn,
+            )?,
+        );
+
+        if amount == account.lamports {
+            // Claim to get whole account
+            instructions.push(claim(
+                &spl_stake_pool::id(),
+                &pool,
+                &pool_withdraw_authority,
+                &stake_pubkey,
+                &config.owner.pubkey(),
+                &burn_from,
+                &pool_data.pool_mint,
+                &spl_token::id(),
+                &stake_program_id(),
+            )?);
         } else {
-            println!("{:<44} Unsupported account data format", address);
+            // Withdraw to split account
+            let stake_receiver: Pubkey = match stake_receiver {
+                Some(value) => *value,
+                None => {
+                    // Account for tokens not specified, creating one
+                    println!(
+                        "Creating account to receive stake {}",
+                        stake_receiver_account.pubkey()
+                    );
+
+                    let stake_receiver_accont_balance = config
+                        .rpc_client
+                        .get_minimum_balance_for_rent_exemption(STAKE_STATE_LEN)?;
+
+                    instructions.push(
+                        // Creating new account
+                        system_instruction::create_account(
+                            &config.fee_payer.pubkey(),
+                            &stake_receiver_account.pubkey(),
+                            stake_receiver_accont_balance,
+                            STAKE_STATE_LEN as u64,
+                            &stake_program_id(),
+                        ),
+                    );
+
+                    signers.push(&stake_receiver_account);
+
+                    total_rent_free_balances += stake_receiver_accont_balance;
+
+                    stake_receiver_account.pubkey()
+                }
+            };
+
+            instructions.push(withdraw(
+                &spl_stake_pool::id(),
+                &pool,
+                &pool_withdraw_authority,
+                &stake_pubkey,
+                &stake_receiver,
+                &config.owner.pubkey(),
+                &burn_from,
+                &pool_data.pool_mint,
+                &spl_token::id(),
+                &stake_program_id(),
+                amount,
+            )?);
         }
+
+        let mut transaction =
+            Transaction::new_with_payer(&instructions, Some(&config.fee_payer.pubkey()));
+
+        let (recent_blockhash, fee_calculator) = config.rpc_client.get_recent_blockhash()?;
+        check_fee_payer_balance(
+            config,
+            total_rent_free_balances + fee_calculator.calculate_fee(&transaction.message()),
+        )?;
+        unique_signers!(signers);
+        transaction.sign(&signers, recent_blockhash);
+
+        return Ok(Some(transaction));
     }
+
+    println!(
```

Comment:
Instead of failing here, the next step should be to go back through the accounts, claim small accounts that add up to the withdraw amount, then merge them all into the receiver account

Reply:
Yes, this is my plan to do it this way as well, however I thought it would be better to have at least the basic CLI now and improve it gradually.

Reply:
Ah ok, that works!